### PR TITLE
Implement Advanced Error Presentations

### PR DESCRIPTION
**Pull request by asbng**
This Pull Request implements first measures to advance with #144.

The changes improve error presentation and readability for UIX users.
Common errors and possible solutions can be represented by throwing `KnownError`s as defined in `src/types/errors.ts`.

### CODE REVIEW

**Comment by jonasstrehle** [comment]
We should consider that that might collide with "unhandledrejection" https://github.com/unyt-org/datex-core-js-legacy/blob/0e3224c02692cd9f10e84e6b4342e2aa9a146dc9/init.ts#L22

**Comment by asbng** [comment]
Indeed, this will definitely be problematic as the current implementation in this PR exits Deno immediately whereas the `unhandledrejection` handler in datex-core-legacy initiates asynchronous work. We might want to somehow integrate / unify the solutions with each other.

Also, https://github.com/unyt-org/datex-core-js-legacy/blob/0e3224c02692cd9f10e84e6b4342e2aa9a146dc9/init.ts#L26 should not be executed when the UIX CLI encounters an error as it performs error presentation on its own.

**Comment by benStre** [comment]
They added a fix for the websocket exception in May (https://github.com/denoland/deno/pull/21849), not sure if there are any other promise rejections that cannot be handled, but I think we can remove the `unhandledrejection` handler for now.

File: src/types/errors.ts
Comment (benStre) [inline src/types/errors.ts:1]: I would assume that there are only typescript type definitions in a directory called "types", not actual class implementations 😄. Maybe we should create a different directory for this case or move errors.ts somewhere else?
@asbng @jonasstrehle

Comment (asbng) [reply src/types/errors.ts:1]: Feel free to put it somewhere else! I wasn't sure where to put such a globally relevant class either.

Comment (benStre) [reply src/types/errors.ts:1]: Okay, let's put it in the app folder then.

Comment (asbng) [reply src/types/errors.ts:1]: Okay, I moved `errors.ts`.

**Comment by asbng** [comment]
I hooked up error reporting via `sendReport()` from `datex-core-legacy`. This will only work correctly, however, once unyt-org/datex-core-js-legacy#107 is addressed.

**Review by benStre — APPROVED** [review]

**Pull request merged**